src/OSPSuite.Presentation/Services/ParameterIdentificationTask.cs

```diff
@@ -242,6 +242,34 @@ public bool SimulationCanBeUsedForIdentification(ISimulation simulation)
          return _simulationSelector.SimulationCanBeUsedForIdentification(simulation);
       }
 
+      public void UpdateParameterIdentificationsUsing(IEnumerable<DataRepository> observedData)
+      {
+         var updatedMappingsInfo = new List<(string ParameterName, string OutputPath)>();
+         observedData.Each(data =>
+         {
+            ParameterIdentificationsUsingObservedData(data).Each(parameterIdentification =>
+            {
+               parameterIdentification.OutputMappingsUsingDataRepository(data).Each(outputMapping =>
+               {
+                  var existingDataCount = outputMapping.WeightedObservedData?.Count ?? 0;
+                  var newDataCount = data.BaseGrid.Count;
+                  if (existingDataCount != newDataCount)
+                  {
+                     outputMapping.WeightedObservedData = new WeightedObservedData(data);
+                     updatedMappingsInfo.Add((parameterIdentification.Name, outputMapping.FullOutputPath));
+                     _executionContext.PublishEvent(new WeightObservedDataChangedEvent(outputMapping));
+                  }
+               });
+            });
+         });
```

Comment:
To reduce nesting and increase readability, can you refactor some methods from these nested iterations